lib/components/mixins/tapable.dart

```diff
@@ -3,71 +3,78 @@ import 'dart:ui';
 import 'package:flutter/foundation.dart';
 import 'package:flutter/gestures.dart';
 
+import '../component.dart';
+import '../../extensions/offset.dart';
+import '../../extensions/vector2.dart';
 import '../../game/base_game.dart';
-import '../position_component.dart';
 
-mixin Tapable on PositionComponent {
-  void onTapCancel() {}
-  void onTapDown(TapDownDetails details) {}
-  void onTapUp(TapUpDetails details) {}
+mixin Tapable on Component {
+  bool onTapCancel() {
```

Comment:
I didn't get it why you have made this method return a `bool`, I didn't noticed this return being used on the new code, am I missing something?

Reply:
It is used to determine that the call has been handled and shouldn't propagate further, in the example of taps it could be several components on top of each other and if only the top one should handle it then the implementer will return false and if it want it to go through the whole tree, or until some condition is met, it should return true.

Reply:
I see, I imagined it was the reason, but I can't find where this bubbling prevention code is, it seems to me that the return of those methods are not being checked anywhere.

Reply:
It is, check in the `propagateToChildren` method.

Reply:
oh, nice, I missed that, but shouldn't it be on the gesture detector as well? So if one component handles the tap, it should not be bubbled to other components?

Reply:
is that desirable? I would expect if I click on a place with several things, all things to be clicked.

Reply:
The implementer chooses, the standard is that you have all clicked.
I was thinking that if you for example only would want to interact with the top component you'd return false, but the check should be done on the reversed priority list then, haven't done that yet.

Reply:
I see, it's optional as you can simply return true. makes sense to me